docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,137 @@
+- Launch the pod/container.
+
+There are several issues here:
+
+- The default capability set heavily depends on Docker's implementation, Kubernetes has no control over that.
+When Docker changes the default capability set across versions, the effective set of default capabilities can change suddenly.
+- To implement the same semantics as today's Docker runtime, other runtimes (e.g. rkt) are required to depend on the implicit Docker default capability set as well. And if Docker changes the default capability set, then consistency is not guaranteed.
+- From a security point of view, an explicit capability set which does not rely on undefined defaults is preferable for auditing purposes.
+
+## Solution
+
+The proposed solution will:
+- Keep the k8s API as is today. Users will not need to change anything.
+- Define a default capability whitelist inside k8s, and document it.
+- Add a `DefaultCapabilities` field to the `PodSecurityPolicySpec` which overrides the default capability whitelist inside k8s.
+- Set the default capability whitelist in the `PodSpec`'s `SecurityContext`.
+- Make CRI expose the whitelist based capability interface, and put the burden on the runtime to implement that.
+- Make the default capability set **visible** through the container spec, for example:
+
+```shell
+$ kubectl get pod -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+  ...
+  spec:
+    containers:
+    - command:
+      - sleep
+      - "1000"
+      image: busybox
+      imagePullPolicy: Always
+      name: my-busybox
+      securityContext:
+        capabilities:
+          default:
+          - SETPCAP, MKNOD, AUDIT_WRITE, ...
+          add:
+          - NET_ADMIN
+          drop:
+          - MKNOD
+      ...
+  status:
+    containerStatuses:
+    - image: busybox
+      name: my-busybox
+      state:
+        running:
+          startedAt: 2016-09-27T22:33:38Z
+    ...
+```
+
+The `PodSecurityPolicySpec` gets a new `DefaultCapabilities` field. The existing fields `DefaultAddCapabilities`, and `RequiredDropCapabilities` will be marked as deprecated.
+
+```
+type PodSecurityPolicySpec struct {
+...
+        // DefaultAddCapabilities is the default set of capabilities that will be added to the container
+        // unless the pod spec specifically drops the capability. You may not list a capability in both
+        // DefaultAddCapabilities and RequiredDropCapabilities.
+        // Note: this field is deprecated in favor of the DefaultCapabilities field.
+        DefaultAddCapabilities []api.Capability `json:"defaultAddCapabilities,omitempty"`
+        // RequiredDropCapabilities are the capabilities that will be dropped from the container.  These
+        // are required to be dropped and cannot be added.
+        // Note: this field is deprecated in favor of the DefaultCapabilities field.
+        RequiredDropCapabilities []api.Capability `json:"requiredDropCapabilities,omitempty"`
```

Comment:
What happens when both `DefaultCapabilities` and `Required{Add,Drop}Capabilities` are set? I think they should all be merged.